File: Seccion 11 - Funciones en C++/Seccion_11_Python/18_parametros_matriz.py

import numpy as np

m = np.array([[1, 2, 3], [4, 5, 6]])

def mostrar_matriz(estado=""):
    
    print('\nMostrando matriz {}:\n'.format(estado))
    
    for i in range(m.shape[0]):
        for j in range(m.shape[1]):
            print(m[i][j], end=" ")
        print()

def calcular_cuadrado():
    global m
    
    for i in range(m.shape[0]):
        for j in range(m.shape[1]):
            m[i][j] *= m[i][j]

mostrar_matriz(estado="original")
calcular_cuadrado()
mostrar_matriz(estado="elevada")

File: Seccion 11 - Funciones en C++/Seccion_11_Dart/18_parametros_matriz.dart

import 'dart:io';

void mostrarMatriz(List<List> m, [String estado='']) {
  print('\nMostrando matriz $estado:\n');

  for (int i = 0; i < m.length; i++) {
    for (int j = 0; j < m[0].length; j++) {
      stdout.write('${m[i][j]} ');
    }
    print('');
  }
}

void calcularCuadrado(List<List> m) {
  for (int i = 0; i < m.length; i++) {
    for (int j = 0; j < m[0].length; j++) {
      m[i][j] *= m[i][j];
    }
  }
}

void main() {

  List<List> m = [[1, 2, 3], [4, 5, 6]];

  mostrarMatriz(m, 'original');
  calcularCuadrado(m);
  mostrarMatriz(m, 'elevada');

}